nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -147,42 +148,13 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
             }
             jg.writeEndArray();
 
-            jg.writeArrayFieldStart("actors");
             final boolean isFetchActors = ctx.getFetched(ENTITY_TYPE).contains(FETCH_ACTORS);
-            for (String actorId : item.getActors()) {
-                if (isFetchActors) {
-                    if (actorId.startsWith(USER_PREFIX + SEPARATOR)) {
-                        actorId = actorId.substring(USER_PREFIX.length() + SEPARATOR.length());
-                        NuxeoPrincipal user = userManager.getPrincipal(actorId);
-                        if (user != null) {
-                            writeEntity(user, jg);
-                            continue;
-                        }
-                    } else if (actorId.startsWith(GROUP_PREFIX + SEPARATOR)) {
-                        actorId = actorId.substring(GROUP_PREFIX.length() + SEPARATOR.length());
-                        NuxeoGroup group = userManager.getGroup(actorId);
-                        if (group != null) {
-                            writeEntity(group, jg);
-                            continue;
-                        }
-                    } else {
-                        NuxeoPrincipal user = userManager.getPrincipal(actorId);
-                        if (user != null) {
-                            writeEntity(user, jg);
-                            continue;
-                        } else {
-                            NuxeoGroup group = userManager.getGroup(actorId);
-                            if (group != null) {
-                                writeEntity(group, jg);
-                                continue;
-                            }
-                        }
-                    }
-                }
-                jg.writeStartObject();
-                jg.writeStringField("id", actorId);
-                jg.writeEndObject();
-            }
+            jg.writeArrayFieldStart("actors");
+            writeActors(item.getActors(), isFetchActors, jg);
+            jg.writeEndArray();
```

Comment:
This is refactoring which is nice but can you please make a separate commit to isolate it from the changes purely related to the reassignment issue.

Reply:
Yes, I can!
In fact, it is not an issue, but a feature/improvement. That's the reason why I didn't consider the separation.
But, in fact, it seems to be a good candidate to be back-ported, so I'll take your suggestion. Thanks!

Reply:
OK, so finally, should this part be of the "Small improvements and code cleanup" commit?

Reply:
Not totally. The idea of extracting to a method was to avoid code repetition, since actors and delegated actors are written in the same way.
The second commit has some refactor of code to be (hopefully!) cleaner and easy to understand.
Do you think that it makes sense this way? Thanks!

Reply:
Totally